doc/src/contracts/gateway.md

```diff
@@ -0,0 +1,46 @@
+# Gateway
+
+The name `gateway` used in this documentation refers to those entities which reside
+on axelar chain, which can also be called internal gateways. On the other hand we have
+external gateways, which are on the external chains that axelar supports. 
+
+Gateway is how messages enter amplifier. Here are the steps taken throughout the lifecycle of 
+a message:
```

Comment:
We probably want to add the step of the user sending a message to the external gateway.